lib/pool.js

```diff
@@ -40,6 +41,10 @@ class Pool extends PoolBase {
       throw new InvalidArgumentError('invalid connections')
     }
 
+    if (reset != null && reset.constructor !== Boolean) {
```

Comment:
Why not `typeof reset !== 'boolean'`?

Reply:
Was mostly about a way of working, but reverted it to keep in sync with previous @KhafraDev suggestion as well 👍